### ISSUE TITLE
[sort-type-union-intersection-members] Autofixed type not equivalent to original type

### ISSUE THREAD

**Issue by ilyub** [typescript-eslint/typescript-eslint#4581]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/sort-type-union-intersection-members": "warn"
  }
}
```

```TS
type Expected = (new (x: string) => object) &
  ((x: number) => boolean) & {
    readonly x: number;
    readonly y: string;
  };
```

**Expected Result**

Should be fixed to:
```ts
type Expected = ((x: number) => boolean) &
  (new (x: string) => object) & {
    readonly x: number;
    readonly y: string;
  };
```

**Actual Result**

Got:
```ts
type Expected = ((x: number) => boolean) & new (x: string) => object & {
  readonly x: number;
  readonly y: string;
};
```

**Additional Info**

No

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.12.1` |
| `@typescript-eslint/parser`        | `5.12.1` |
| `TypeScript`                       | `4.5.5` |
| `ESLint`                           | `8.9.0` |
| `node`                             | `14.18.0` |

**Comment by islandryu** [comment]
I think this issue has a bug where the parentheses are removed in constructorType, and a bug where the line breaks don't work.
I fixed the former in #4590.

**Comment by JoshuaKGoldberg** [comment]
Hmm, do you have a repro case for the line breaks version?
